src/DotNet.Glob/Evaluation/LiteralTokenEvaluator.cs

```diff
@@ -20,6 +22,12 @@ public bool IsMatch(string allChars, int currentPosition, out int newPosition)
                 var compareChar = _token.Value[counter];
                 var currentChar = allChars[newPosition];
 
+                if (_caseInsensitive)
```

Comment:
I _think_ that adding this additional `if statement` in, shouldn't have any noteable performance implications, because, ad the _caseInsensitive flag will either stay as `true` or `false` and won't change, [CPU branch prediction should kick in relatively nicely to optimise the additional check away](http://igoro.com/archive/fast-and-slow-if-statements-branch-prediction-in-modern-processors/).

If we wanted to be super careful, we could create `CaseInsensitive` versions of each token, i.e `CaseInsensitiveLiteralTokenEvaluator`, `CaseInsensitiveLiteralTokenEvaluator` - and then use those tokens instead. Not sure whether I like that design or not though. What do you think?

Reply:
I know that lightning performance one of this library's main goals, but the performance impliciations of adding/avoiding a boolean check (even if invoked once per char in a literal token) is practically non-existent. Really not a fan of premature optimizations. I'd be surprised if you'd notice any difference in benchmarks (haven't checked your benchmarks, though! :wink:)

Reply:
*If* you wanted to avoid it, how about setting an private delegate in the constructur that gets the character?

Something like:

    private Func<char, char> _getChar;

    public LiteralTokenEvaluator(LiteralToken token, bool caseInsensitive)
    {
        _token = token;
        if (caseInsensitive) { // at parse time:
            _getChar = c => char.ToLowerInvariant(c); // gets lower invariant
        }
        else {
            _getChar = c => c; // char as-is
        }
    }

And in `IsMatch` method in the while loop:

     var compareChar = _getChar(_token.Value[counter]);
     var currentChar = _getChar(allChars[newPosition]);

Though I rarely deal with optimizations at this level, so not entirely sure whether this will in fact be an improvement or the opposite....

Reply:
Just deleted my second comment, didn't make sense. It's 12PM here, not thinking straight ;-)